data/hai/hai reveal 4 middle.txt

```diff
@@ -658,13 +658,53 @@ mission "Pirate Troubles [4]"
 
 
 
+# Blame Amazinite for this. They can proc at the same time if you fly around a bunch, but generally shouldn't.
+mission "Hai Reveal [A03b]"
+	invisible
+	source
+		government "Republic" "Syndicate" "Hai" "Free Worlds" "Independent"
+		attributes "spaceport"
+	to offer
+		# You will only trigger this before A04 if you've already done Pirate Troubles.
+		has "Hai Reveal [A03a] Symbol Identification: offered"
+		or
+			has "Pirate Troubles [4]: done"
+			has "Pirate Troubles [4]: declined"
+		not "Hai Reveal [A04] Report to Teeneep: offered"
+	on offer
+		dialog `You have already completed the Pirate Troubles mission line which is now part of this story, and are seeing this because you're playing an older save. In it you tracked down Scar's Legion and either took them tribute from the Hai, or destroyed them. Your choices there will be considered in the story here.`
```

Comment:
I think it makes more sense to either 1) rename the missions so that they reoffer or 2) clear the offer conditions on the missions before they're needed so that they can reoffer.

Reply:
Whether or not you like Pirate Troubles, it's been made a more integral part of the HR storyline. It doesn't make sense to just skip over it, as players may not even remember what happened in Pirate Troubles. It is a nearly 3 year old mission line, afterall.

I'd go the mission rename route. Go from `"Pirate Troubles [X]"` to something like `"Hai Reveal: Pirate Troubles [X]"`, making sure to also cover the conditions.

Reply:
Oh no don't get me wrong, I like it, I just would be irritated playing it twice. That's a me thing though.

I do have a question though - since then saves would already have scar's base added in and these missions would add it again. How would that be handled?

Reply:
Looks like the current event redefines the whole system. We could clean that up to just add the base, then have another event that triggers with the first mission to remove it if it already exists.


```
# Need to remove the planet the asteroid orbits, then readd it without the landable location.
# Trigger on accept of PT [1].
event "clean danoa"
	system "Danoa"
		remove object
			sprite planet/gas6
			distance 2069.42
			period 3240.9
		add 
			sprite planet/gas6
			distance 2069.42
			period 3240.9
			object
				sprite planet/lava1-b
				distance 293
				period 11.3672
			object
				sprite planet/rock0-b
				distance 440
				period 20.9186
			object
				sprite planet/lava2-b
				distance 607
				period 33.895
			object
				sprite "planet/asteroid base"
				distance 809
				period 52.153

# The existing event.
event "scar's hideout"
	system "Danoa"
		remove object
			sprite planet/gas6
			distance 2069.42
			period 3240.9
		add 
			sprite planet/gas6
			distance 2069.42
			period 3240.9
			object
				sprite planet/lava1-b
				distance 293
				period 11.3672
			object
				sprite planet/rock0-b
				distance 440
				period 20.9186
			object
				sprite planet/lava2-b
				distance 607
				period 33.895
			object "Scar's Hideout"
				sprite "planet/asteroid base"
				distance 809
				period 52.153
```